api/v1/models/User.js

```diff
@@ -187,6 +188,17 @@ User.prototype.hasPassword = function(password) {
     });
 };
 
+User.prototype.updateContactInfo = function(newEmail) {
+  if(!_.isNull(this.get('password'))) {
+    const message = 'Cannot update the contact info of a Basic user';
+    throw new UnprocessableRequestError(message);
```

Comment:
Models shouldn't really be throwing request errors (otherwise they have implied knowledge of a 'request'). This should probably all happen in the service method that takes a user and a new email.